src/main/java/org/opensearch/flowframework/indices/FlowFrameworkIndicesHandler.java

```diff
@@ -628,13 +712,38 @@ public void updateFlowFrameworkSystemIndexDoc(
             listener.onFailure(new FlowFrameworkException(errorMessage, RestStatus.BAD_REQUEST));
         } else {
             try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-                UpdateRequest updateRequest = new UpdateRequest(WORKFLOW_STATE_INDEX, documentId);
-                XContentBuilder builder = XContentFactory.jsonBuilder();
-                updatedDocument.toXContent(builder, null);
-                updateRequest.doc(builder);
-                updateRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-                updateRequest.retryOnConflict(RETRIES);
-                client.update(updateRequest, ActionListener.runBefore(listener, context::restore));
+                UpdateDataObjectRequest updateRequest = UpdateDataObjectRequest.builder()
+                    .index(WORKFLOW_STATE_INDEX)
+                    .id(documentId)
+                    .tenantId(tenantId)
+                    .dataObject(updatedDocument)
+                    .retryOnConflict(RETRIES)
+                    .build();
+                sdkClient.updateDataObjectAsync(updateRequest).whenComplete((r, throwable) -> {
+                    context.restore();
+                    if (throwable == null) {
+                        UpdateResponse response;
+                        try {
+                            response = UpdateResponse.fromXContent(r.parser());
+                            logger.info("Deleted workflow state doc: {}", documentId);
```

Comment:
Aren't we updating here?

Reply:
Already stumbled on this and changed it on my local.